libcudacxx/include/cuda/std/detail/libcxx/include/__cuda/ptx/ptx_helper_functions.h

```diff
@@ -0,0 +1,47 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef PTX_HELPER_FUNCTIONS_H_
+#define PTX_HELPER_FUNCTIONS_H_
+
+_LIBCUDACXX_BEGIN_NAMESPACE_CUDA_PTX
+
+// Private helper functions
+inline _LIBCUDACXX_DEVICE _CUDA_VSTD::uint32_t __as_smem_ptr(const void* __ptr)
```

Comment:
question: This is hard to disambiguate, because the difference is in the middle of the name. Could we use a name that has identical prefix and only differs in the end, e.g. `__as_ptr_smem` and `__as_ptr_dsmem`

Reply:
I am fine either way. I can imagine your suggestion is easier to disambiguate. Changed.